libsolidity/interface/Natspec.cpp

```diff
@@ -78,7 +78,7 @@ Json::Value Natspec::userDocumentation(ContractDefinition const& _contractDef)
 				doc["methods"][it.second->externalSignature()]["notice"] = value;
 		}
 
-	for (auto const& event: _contractDef.definedInterfaceEvents())
+	for (auto const& event: _contractDef.interfaceEvents())
```

Comment:
But it's a set of `EventDefinition const*` with `CompareByID`. Different event definitions will get different AST IDs so I don't think this will limit the choice to only one of them.

In the end, this is something that should be easy to see in tests, without speculating. Please add a test in `SolidityNatspecJSON.cpp` with an event defined in a library and in the contract, where both have Natspec.

Might also be a good idea to cover more corner cases (if you can incorporate some of them without making these extra tests too verbose):
- Event defined in two libraries, but not in the contract.
- Event defined in a library and an inherited contract (rather than current contract).
- The cases above where these events are defined but not all of them actually have Natspec.

Reply:
I've added the tests and will look why it seems to be working as expected, even though the AST IDs should be different.